lib/CodeGen/CGCall.cpp

```diff
@@ -2819,6 +2818,25 @@ void CodeGenFunction::EmitCallArg(CallArgList &args, const Expr *E,
     return;
   }
 
+  if (type->isSamplerT() && CGM.getLangOpts().CLSamplerOpaque) {
```

Comment:
why do we need this? shouldn't all int to sampler casts be handled by CGExprConstant.cpp and CGExprScalar.cpp?